README.md

```diff
@@ -0,0 +1,82 @@
+# CrimLog API
+
+The CrimLog API is the backbone of the CrimLog protocol. It is a [GraphQL](https://graphql.org/) API written in [TypeScript](https://www.typescriptlang.org/) with the [Nest.js](http://nestjs.com/) framework. [Prisma](https://prisma.io/) is used as the ORM and [pnpm](https://pnpm.io/) as the package manager.
+
+# Project Setup
+
+## Prerequisites
+
+-   Node.js v16.18 or greater (https://nodejs.org/en/download/releases/)
+-   pnpm v7.x (https://pnpm.io/installation)
+-   Git (https://git-scm.com/downloads)
+-   MongoDB (https://www.mongodb.com/docs/manual/tutorial/getting-started/)
+
+## Install Dependencies
+
+This is a [pnpm](https://pnpm.io/) project, and the usage of other package managers such as npm or yarn is strongly discouraged. After installing Node.js, pnpm can be enabled using the built-in [Corepack](https://nodejs.org/api/corepack.html) API. Run the following commands to setup pnpm on your system:
+
+```cmd
+corepack enable
+corepack prepare pnpm@latest --activate
+```
+
+Once pnpm has been enabled on your system, navigate to the root directory and execute the command:
+
+```cmd
+pnpm install --frozen-lockfile
+```
+
+This command locally installs all the project dependencies may take some time to run.
+
+Afterwards, setup a template `.env` file by copying the contents of the `.env.sample` file into a new `.env` file located in the root.
+
+## Database
+
+[MongoDB](https://www.mongodb.com/) is the API's database provider. Once you've created your own MongoDB instance, obtain the connection string and set that as the value of the `DATABASE_URL` environment variable.
+
+To seed your database with some sample data, first run the ` prisma:push` script. This will apply the Prisma schema to your newly-created Mongo database. Then execute the `prisma:seed` script to populate the database.
+
+```cmd
+pnpm prisma:push
+pnpm prisma:seed
+```
+
+## Auto-generated Code
+
+CrimLog makes use of several code generation tools to improve the development experience. It is important to regenerate files after making changes to certain areas of the codebase.
+
+### Prisma
+
+[Prisma](https://prisma.io/), an ORM, provides complete and thorough TypeScript types for all database models and queries. The `schema.prisma` file is the single source of truth for these types.
+
+Whenever the `schema.prisma` is updated, the `prisma:generate` script will need to be run to regenerate the Prisma types. Alternativel, `prisma:generate:w` can be run once to continuously watch the Prisma schema file and regenerate types automatically on save.
+
+```cmd
+pnpm prisma:generate
```

Comment:
Rearrange this section with Auto-Generated Code - Prisma section has to be before the seed - 
pnpm prisma:generate 
must be run before
pnpm prisma:seed

Reply:
After running all of this Git is noticing changes in graphql/typings.ts as a changed file but not really showing changes - don't think it's a problem but just wanted to make a note

Reply:
> After running all of this Git is noticing changes in graphql/typings.ts as a changed file but not really showing changes - don't think it's a problem but just wanted to make a note

good to know, this might be because of the EOL sequence. I can perform brief debugging on your machine when you're available

Reply:
> Rearrange this section with Auto-Generated Code - Prisma section has to be before the seed - pnpm prisma:generate must be run before pnpm prisma:seed

resolved in 91d13a9